test/metabase/server/middleware/session_test.clj

```diff
@@ -397,7 +397,54 @@
 
 ;;; ----------------------------------------------------- Session timeout -----------------------------------------------------
 
-(deftest session-timeout-tests
+(deftest session-timeout-validation-test
+  (testing "Setting the session timeout should fail if the timeout isn't positive"
+    (is (thrown-with-msg?
+         java.lang.Exception
+         #"Session timeout amount must be positive"
+         (mw.session/session-timeout! {:unit "hours", :amount 0})))
+    (is (thrown-with-msg?
+         java.lang.Exception
+         #"Session timeout amount must be positive"
+         (mw.session/session-timeout! {:unit "minutes", :amount -1}))))
+  (testing "Setting the session timeout should fail if the timeout is too large"
+    (is (thrown-with-msg?
+         java.lang.Exception
+         #"Session timeout must be less than 100 years"
+         (mw.session/session-timeout! {:unit "hours", :amount (* 100 365.25 24)})))
+    (is (thrown-with-msg?
+         java.lang.Exception
+         #"Session timeout must be less than 100 years"
+         (mw.session/session-timeout! {:unit "minutes", :amount (* 100 365.25 24 60)}))))
+  (testing "Setting the session timeout shouldn't fail if the timeout is between 0 and 100 years exclusive"
+    (is (some? (mw.session/session-timeout! {:unit "minutes", :amount 1})))
+    (is (some? (mw.session/session-timeout! {:unit "hours", :amount 1})))
+    (is (some? (mw.session/session-timeout! {:unit "minutes", :amount (dec (* 100 365.25 24 60))})))
+    (is (some? (mw.session/session-timeout! {:unit "hours", :amount (dec (* 100 365.25 24))})))))
+
+(deftest session-timeout-env-var-validation-test
+  (let [set-and-get (fn [timeout]
+                      (mt/with-temp-env-var-value [mb-session-timeout (json/generate-string timeout)]
+                        (mw.session/session-timeout)))]
+    (testing "Setting the session timeout with env var should work with valid timeouts"
+      (doseq [timeout [{:unit "hours", :amount 1}
+                       {:unit "hours", :amount (dec (* 100 365.25 24))}]]
+        (is (= timeout
+               (set-and-get timeout)))))
+    (testing "Setting the session timeout via the env var should fail if the timeout isn't positive"
+      (doseq [amount [0 -1]
+              :let [timeout {:unit "hours", :amount amount}]]
+        (is (nil? (set-and-get timeout)))
+        (is (= (println-str "WARNING: Session timeout amount must be positive.")
```

Comment:
We have a helper called something like `mt/log-messages-for-level` to capture log messages in tests